airflow/jobs/scheduler_job.py

```diff
@@ -521,14 +521,24 @@ def _process_executor_events(self, session: Session = None) -> int:
                 )
                 self.log.error(msg, ti, state, ti.state, info)
 
-                request = TaskCallbackRequest(
-                    full_filepath=ti.dag_model.fileloc,
-                    simple_task_instance=SimpleTaskInstance(ti),
-                    msg=msg % (ti, state, ti.state, info),
-                )
-                self.log.info('Setting task instance %s state to %s as reported by executor', ti, state)
-                ti.set_state(state)
-                self.processor_agent.send_callback_to_execute(request)
+                # Get task from the Serialized DAG
+                try:
+                    dag = self.dagbag.get_dag(ti.dag_id)
+                    task = dag.get_task(ti.task_id)
+                except Exception as ex:
+                    self.log.exception("Marking task instance %s as failed. Reason: %s", ti, ex)
+                    ti.set_state(state)
+                    continue
+                ti.task = task
+                if task.on_retry_callback or task.on_failure_callback:
```

Comment:
@kaxil Is this right for serialized tasks?

Reply:
Yeah, we currently store it as string:

```python
(Pdb) pp SerializedDAG.serialize_dag(dag)
{'_dag_id': 'test_process_executor_events_with_callback',
 '_task_group': {'_group_id': None,
                 'children': {'dummy_task': (<DagAttributeTypes.OP: 'operator'>,
                                             'dummy_task')},
                 'downstream_group_ids': [],
                 'downstream_task_ids': [],
                 'prefix_group_id': True,
                 'tooltip': '',
                 'ui_color': 'CornflowerBlue',
                 'ui_fgcolor': '#000',
                 'upstream_group_ids': [],
                 'upstream_task_ids': []},
 'dag_dependencies': [],
 'edge_info': {},
 'fileloc': '/test_path1/',
 'params': {},
 'schedule_interval': {<Encoding.TYPE: '__type'>: <DagAttributeTypes.TIMEDELTA: 'timedelta'>,
                       <Encoding.VAR: '__var'>: 86400.0},
 'start_date': 1451606400.0,
 'tasks': [{'_downstream_task_ids': [],
            '_inlets': [],
            '_is_dummy': True,
            '_outlets': [],
            '_task_module': 'airflow.serialization.serialized_objects',
            '_task_type': 'SerializedBaseOperator',
            'label': 'dummy_task',
            'on_failure_callback': '            task1 = '
                                   'DummyOperator(task_id=task_id_1, '
                                   'on_failure_callback=lambda x: '
                                   'print("hi"))\n',
            'pool': 'default_pool',
            'task_id': 'dummy_task',
            'template_ext': [],
            'template_fields': [],
            'template_fields_renderers': {},
            'ui_color': '#e8f7e4',
            'ui_fgcolor': '#000'}],
 'timezone': 'UTC'}
```

or 

```python
(Pdb) from airflow.serialization.serialized_objects import SerializedBaseOperator, SerializedDAG
(Pdb) pp SerializedDAG.serialize_dag(dag)
{'_dag_id': 'test_process_executor_events_with_callback',
 '_task_group': {'_group_id': None,
                 'children': {'dummy_task': (<DagAttributeTypes.OP: 'operator'>,
                                             'dummy_task')},
                 'downstream_group_ids': [],
                 'downstream_task_ids': [],
                 'prefix_group_id': True,
                 'tooltip': '',
                 'ui_color': 'CornflowerBlue',
                 'ui_fgcolor': '#000',
                 'upstream_group_ids': [],
                 'upstream_task_ids': []},
 'dag_dependencies': [],
 'edge_info': {},
 'fileloc': '/test_path1/',
 'params': {},
 'schedule_interval': {<Encoding.TYPE: '__type'>: <DagAttributeTypes.TIMEDELTA: 'timedelta'>,
                       <Encoding.VAR: '__var'>: 86400.0},
 'start_date': 1451606400.0,
 'tasks': [{'_downstream_task_ids': [],
            '_inlets': [],
            '_is_dummy': True,
            '_outlets': [],
            '_task_module': 'airflow.serialization.serialized_objects',
            '_task_type': 'SerializedBaseOperator',
            'label': 'dummy_task',
            'on_retry_callback': '            task1 = '
                                 'DummyOperator(task_id=task_id_1, '
                                 'on_retry_callback=lambda x: print("hi"))\n',
            'pool': 'default_pool',
            'task_id': 'dummy_task',
            'template_ext': [],
            'template_fields': [],
            'template_fields_renderers': {},
            'ui_color': '#e8f7e4',
            'ui_fgcolor': '#000'}],
 'timezone': 'UTC'}
(Pdb)
```